observability/observability.go

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observability
+
+import (
+	"context"
+
+	"google.golang.org/grpc/grpclog"
+)
+
+var (
+	logger = grpclog.Component("observability")
+)
+
+// Start is the opt-in API for gRPC Observability plugin.
+//   - it loads observability config from environment;
+//   - it registers default exporters if not disabled by the config;
+//   - it sets up binary logging sink against the logging exporter.
```

Comment:
I'm not totally sure about this pattern.

This is setting a global that must happen before any grpc clients/servers are created.  We should document this.  If they have an import that creates a grpc client/server in an `init` then that could be a problem.  Maybe the `binarylog` API should use an atomic for `SetSink` so it's at least not a potential panic.

Also, since we can only have one (global) binary logger registered, then the user must not use their own binary logger implementation if they use this.

@menghanl, thoughts on this?

Reply:
When do we expect the users to not call `Start()`?

And if `End()` is global, how do they call it, since it will disable observability for all the clients/servers? Or is it more like a `Flush()`?

Reply:
> When do we expect the users to not call Start()?

Not call `Start()`? If users don't want to use the observability features.

If you mean when to call `Start()`, I'm not sure about the best practice here. `binarylog`'s documentation says users should only call its API in `init()`. I propose to ask users call it in `main()` for 2 reasons:
- The plugin only need to register to the hook points (in this case, sink) before any gRPC clients/servers are created.
- Enabling the plugin means non-trivial performance cost, by placing them in `main()`, users will have to make an explicit choice for every binary.

> And if End() is global, how do they call it, since it will disable observability for all the clients/servers? Or is it more like a Flush()?

Functional-wise, I'm worried about data loss in the process exit phase. So, I hope to add a `Flush()`-like call. On the other hand, I found this can be a good opportunity to clean-up resources (underlying clients, goroutines, stuff), and updated its name to `End()`.

I agree we should include more documentation here. The conflicts in `init` and `binarylog` are risks. Is there any other places I should update (like: https://github.com/grpc/grpc-go/tree/master/Documentation)? This effort might not be ready enough for examples yet.